extensions/circuit/src/adapters/allocate_frame.rs

```diff
@@ -76,19 +76,26 @@ pub struct AllocateFrameWriteRecord<F> {
 pub struct AllocateFrameAdapterColsWom<T> {
     pub from_state: ExecutionState<T>,
     pub from_frame: FrameState<T>,
-    pub target_reg_ptr: T,  // Pointer to target register
-    pub allocation_size: T, // amount_imm field from instruction (b)
-    pub src_reg_ptr: T,     // if reading from register
-    pub result: T,
-    pub read_imm_or_reg: T,
+    pub amount_reg: T,
+    // amount from register
+    pub amount: [T; RV32_REGISTER_NUM_LIMBS],
+    // immediate amount
+    pub amount_imm: T,
+    // 0 if imm, 1 if reg
+    pub amount_imm_or_reg: T,
+    // new frame pointer: provided by the prover
+    pub next_frame_ptr: [T; RV32_REGISTER_NUM_LIMBS],
```

Comment:
I think this could be a single T?

Reply:
For now I left wom.reads/writes working as memory was, so reads and writes work on RV32_REGISTER_NUM_LIMBS.
Maybe we do these optimizations (exploiting the full F in wom) in a new PR?